test/e2e/10_helm_chart.bats

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bats
+
+load lib/env
+load lib/install
+load lib/poll
+
+function setup() {
+  kubectl create namespace "$E2E_NAMESPACE"
+  install_git_srv
+  install_tiller
+  install_helm_operator_with_helm
+  kubectl create namespace "$DEMO_NAMESPACE"
+}
+
+@test "Helm chart installation smoke test" {
+  # The gitconfig secret must exist and have the right value
+  poll_until_equals "gitconfig secret" "${GITCONFIG}" "kubectl get secrets -n ${E2E_NAMESPACE} gitconfig -ojsonpath={..data.gitconfig} | base64 --decode"
+
+  poll_until_true "namespace $DEMO_NAMESPACE" "kubectl describe ns/$DEMO_NAMESPACE"
+
+  # Apply the HelmRelease fixtures
+  kubectl apply -f "$FIXTURES_DIR/releases" >&3
+
+  poll_until_true 'podinfo-helm-repository HelmRelease' 'kubectl -n demo get helmrelease/podinfo-helm-repository | grep -e "deployed"'
+  poll_until_true 'podinfo-git HelmRelease' 'kubectl -n demo describe helmrelease/podinfo-git | grep -e "deployed"'
```

Comment:
Doesn't this simply test that the releases where applied? Shouldn't we test that the operator applied them instead?

Reply:
No, it tests that the releases have been made by the operator and were successful, hence the `| grep -e "deployed"`.

Reply:
Ah, sorry, I missed that. It would be cleaner to use `poll_until_equals` though and extracting the status field.

Reply:
I forgot about the existence of that method while modifying what I copied from Flux. Will use that instead, much better indeed.